geode-core/src/distributedTest/java/org/apache/geode/distributed/internal/ClusterDistributionManagerDUnitTest.java

```diff
@@ -465,6 +470,26 @@ public void testWaitForViewInstallationDisconnectDS()
     future.get(getTimeout().toMillis(), TimeUnit.MILLISECONDS);
   }
 
+  @Test
+  public void testMemberJoinsAfterNewViewIsInstalled() {
+    vm1.invoke("join the cluster", () -> getSystem().getDistributedMember());
+    vm1.invoke(
+        () -> system.getDistributionManager().addMembershipListener(new MembershipListener() {
+          @Override
+          public void memberJoined(DistributionManager distributionManager,
+              InternalDistributedMember id) {
+            if (!distributionManager.getDistribution().getMembership().memberExists(id)) {
+              // if MemberJoinedEvent is triggered before the new view is installed,
+              // the test will throw an exception
+              FunctionService.onMember(id)
+                  .execute(new DistributedSystemDUnitTest.FailDeserializationFunction());
```

Comment:
We really need to avoid tangling one dunit test with another dunit test. You should copy FailDeserializationFunction in full from DistributedSystemDUnitTest or extract it to its own file. But please avoid having this dunit test reference anything in another dunit test.